src/lib.rs

```diff
@@ -27,6 +27,15 @@
 #![doc(html_root_url = "https://briansmith.org/rustdoc/")]
 #![cfg_attr(not(feature = "std"), no_std)]
 #![warn(unreachable_pub)]
+#![deny(warnings, missing_docs, clippy::as_conversions)]
```

Comment:
One minor roadblock:

The new combination of `unreachable_pub` and `deny(warnings)` in `lib.rs` seems to be breaking CI for Rust 1.57 in particular based on some (I believe false positive) `unreachable_pub` findings: https://github.com/cpu/webpki/actions/runs/5082891534/jobs/9133182552

I've tried:
* Using `cfg_attr` in `lib.rs` to skip `unreachable_pub` for rustc 1.57 - that doesn't work well because `cfg(version)` is experimental
* Trying to find a `cargo test --all-features` invocation where I can override settings at the CLI so we could gate it in CI. Using `RUSTFLAGS=-A unreachable_pub` or `RUSTFLAGS=-A warnings` hasn't worked. Ditto for `cargo test --all-features -- -A warnings` or `cargo test --all-features -- -A unreachable_pub`.

@djc Any thoughts? :thinking:

Reply:
Maybe just missing some quotes? `RUSTFLAGS="-A unreachable_pub"` seems to do something for me.

Reply:
> Maybe just missing some quotes?

I did use quotes in my testing but didn't put them in my comment, sorry about the confusion.

> RUSTFLAGS="-A unreachable_pub" seems to do something for me.

Are you testing with 1.57? It doesn't error when I provide flags like this but it also doesn't squelch the build errors:

<details>
<summary>Here's what I see trying this w/ 1.57 and this branch:</summary>

```bash

[nix-shell:~/Code/Rust/webpki]$ cargo --version
cargo 1.57.0 (b2e52d7ca 2021-10-21)

[nix-shell:~/Code/Rust/webpki]$ git rev-parse HEAD
0e9408e44bded9b2d2a769a2fe63493a77cd82b0

[nix-shell:~/Code/Rust/webpki]$ RUSTFLAGS="-A unreachable_pub" cargo test
   Compiling rustls-webpki v0.100.1 (/home/daniel/Code/Rust/webpki)
error: unreachable `pub` item
  --> src/subject_name/mod.rs:18:20
   |
18 | pub use dns_name::{DnsNameRef, InvalidDnsNameError};
   | ---                ^^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
note: the lint level is defined here
  --> src/lib.rs:30:9
   |
30 | #![deny(warnings, missing_docs, clippy::as_conversions)]
   |         ^^^^^^^^
   = note: `#[deny(unreachable_pub)]` implied by `#[deny(warnings)]`
   = help: or consider exporting it for use by other crates

error: unreachable `pub` item
  --> src/subject_name/mod.rs:18:32
   |
18 | pub use dns_name::{DnsNameRef, InvalidDnsNameError};
   | ---                            ^^^^^^^^^^^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
   = help: or consider exporting it for use by other crates

error: unreachable `pub` item
  --> src/subject_name/mod.rs:22:1
   |
22 | pub use dns_name::DnsName;
   | ---^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
   = help: or consider exporting it for use by other crates

error: unreachable `pub` item
  --> src/subject_name/mod.rs:25:16
   |
25 | pub use name::{InvalidSubjectNameError, SubjectNameRef};
   | ---            ^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
   = help: or consider exporting it for use by other crates

error: unreachable `pub` item
  --> src/subject_name/mod.rs:25:41
   |
25 | pub use name::{InvalidSubjectNameError, SubjectNameRef};
   | ---                                     ^^^^^^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
   = help: or consider exporting it for use by other crates

error: unreachable `pub` item
  --> src/subject_name/mod.rs:28:22
   |
28 | pub use ip_address::{AddrParseError, IpAddrRef};
   | ---                  ^^^^^^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
   = help: or consider exporting it for use by other crates

error: unreachable `pub` item
  --> src/subject_name/mod.rs:28:38
   |
28 | pub use ip_address::{AddrParseError, IpAddrRef};
   | ---                                  ^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
   = help: or consider exporting it for use by other crates

error: unreachable `pub` item
  --> src/subject_name/mod.rs:31:1
   |
31 | pub use ip_address::IpAddr;
   | ---^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | help: consider restricting its visibility: `pub(crate)`
   |
   = help: or consider exporting it for use by other crates

error: could not compile `rustls-webpki` due to 8 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed

```
</details>

I get the same failure trying other permutations. :dizzy_face:

Reply:
@djc This gross hack to `lib.rs` resolves the problem w/ 1.57: https://github.com/rustls/webpki/pull/72/commits/0c50e66d0133452cfbee9c1f6bd8f5b2cd1fcb78

If you don't like that approach I think it might be best to close this PR and wait until the MSRV is updated to do this cleanup. LMK what you prefer.